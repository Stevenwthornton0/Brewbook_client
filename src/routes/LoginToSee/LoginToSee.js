import React, { Component } from 'react';
import LoginForm from '../../components/LoginForm/LoginForm';

class LoginToSee extends Component {
    static defaultProps = {
        location: {},
        history: {
            push: () => {},
        }
    }
    
    handleLoginSuccess = () => {
        const { location, history } = this.props;
        const destination = (location.state || {}).from || '/';
        history.push(destination)
        window.location.reload();
    }

    render() {
        return (
            <section className='loginPage'>
                <h2>Login <span className='smaller'>to view full brewery details</span></h2>
                <LoginForm
                    onLoginSuccess={this.handleLoginSuccess}
                />
            </section>
        )
    }
}

export default LoginToSee;