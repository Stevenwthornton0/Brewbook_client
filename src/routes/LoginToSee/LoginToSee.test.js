import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import LoginToSee from './LoginToSee';

Enzyme.configure({ adapter: new Adapter() })

describe(`LoginToSee smoke test`, () => {
    it(`renders without crashing`, () => {
        const wrapper = shallow(<LoginToSee />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})