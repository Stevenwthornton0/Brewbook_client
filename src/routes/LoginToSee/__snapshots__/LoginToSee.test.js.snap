// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginToSee smoke test renders without crashing 1`] = `
<section
  className="loginPage"
>
  <h2>
    Login 
    <span
      className="smaller"
    >
      to view full brewery details
    </span>
  </h2>
  <LoginForm
    onLoginSuccess={[Function]}
  />
</section>
`;
